apps/src/codebridge/FileBrowser/prompts/openImportFromBackpackPrompt.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  NewFileFunction,
+  SaveFileFunction,
+} from '@codebridge/codebridgeContext/types';
+import {DEFAULT_FOLDER_ID} from '@codebridge/constants';
+import {validateFileName, getFileNameWithNumberSuffix} from '@codebridge/utils';
+
+import {MultiFileSource, ProjectFile} from '@cdo/apps/lab2/types';
+import {
+  DialogType,
+  DialogControlInterface,
+  extractUserInput,
+} from '@cdo/apps/lab2/views/dialogs';
+import {GenericDropdownProps} from '@cdo/apps/lab2/views/dialogs/GenericDropdown';
+import {BackpackContextType} from '@cdo/apps/sharedComponents/backpack/BackpackAPIContext';
+
+type OpenImportFromBackpackPromptArgsType = {
+  dialogControl: Pick<DialogControlInterface, 'showDialog'>;
+  backpackApi: BackpackContextType;
+  newFile: NewFileFunction;
+  saveFile: SaveFileFunction;
+  projectFiles: MultiFileSource['files'];
+  validationFile?: ProjectFile;
+};
+
+export const openImportFromBackpackPrompt = async ({
+  dialogControl,
+  backpackApi,
+  newFile,
+  saveFile,
+  projectFiles,
+  validationFile,
+}: OpenImportFromBackpackPromptArgsType) => {
+  const handleError = (message: string) => () => {
+    // TODO: send analytics / add logging.
+    console.error(message);
+  };
+  // Delete a file from the backpack.
+  const handleDelete = async (filename: string) => {
+    backpackApi.deleteFiles(
+      [filename],
+      handleError(`Error in deleting file ${filename}`),
+      () => console.log(`Deleted file ${filename}`)
```

Comment:
follow up polish item: indicate to the user that delete was successful?

Reply:
👍  - I'll ask product/design about how to indicate deletion success. Will add to follow-up section.